services/patient-service/patient-service.js

```diff
@@ -142,9 +153,23 @@ class PatientService {
    * @param {mongoose.Types.ObjectId} idNurse Nurse ID
    * @returns {PatientObject} Patient updated
    * @description Set the current value for 'idNurse' of the patient
-   * @throws {NotFoundError} In case Patient was not found
+   * @throws {NotFoundError | BadRequestError}
    */
   async setIdNurse(id, idNurse) {
+    const patients = await this.returnNursePatients(idNurse);
+
+    if (patients.length >= parseInt(process.env.MAX_PATIENTS_FOR_NURSE)) {
+      throw new BadRequestError("This nurse can only have 6 patients or less");
+    }
+
+    const employee = await EmployeeModel.findById(idNurse);
+
+    if (employee.role !== "NURSE") {
+      throw new BadRequestError(
+        "The Employee you're trying to set is not a nurse"
+      );
+    }
+
```

Comment:
Validate first if the nurse exists or if the idNurse correspond to a nurse. After that validate if the nurse has already 6 patients or not.

```suggestion
    const employee = await EmployeeModel.findById(idNurse);

    if (employee.role !== "NURSE") {
      throw new BadRequestError(
        "The Employee you're trying to set is not a nurse"
      );
    }
    
    const patients = await this.returnNursePatients(idNurse);

    if (patients.length >= parseInt(process.env.MAX_PATIENTS_FOR_NURSE)) {
      throw new BadRequestError("This nurse can only have 6 patients or less");
    }
```